server/websocket_test.go

```diff
@@ -1609,6 +1627,42 @@ func TestWSParseOptions(t *testing.T) {
 				}
 				return nil
 			}, ""},
+		{"headers",
```

Comment:
I would add negative tests in the list above and for this one, you have to keep in mind that the fact that we iterate through a map (key/value), the order is not guaranteed. So if you were to run the test multiple times (`go test -race -v -run=TestWSParseOptions ./server -count=100 -vet=off -failfast`), you would get a failure for this test because the resulting string would sometimes be with the `X-Frame-Options: "DENY"` first.

Here is a diff of this file with some corrections:
```
diff --git a/server/websocket_test.go b/server/websocket_test.go
index 0af2d652..6bccf8b7 100644
--- a/server/websocket_test.go
+++ b/server/websocket_test.go
@@ -1480,6 +1480,8 @@ func TestWSParseOptions(t *testing.T) {
                {"bad handshake timeout type", `websocket: { handshake_timeout: [] }`, nil, "unsupported type"},
                {"bad handshake timeout duration", `websocket: { handshake_timeout: "abc" }`, nil, "invalid duration"},
                {"unknown field", `websocket: { this_does_not_exist: 123 }`, nil, "unknown"},
+               {"headers wrong type", `websocket: { headers: 123 }`, nil, "unsupported"},
+               {"headers content wrong type", `websocket: { headers: { 123: 456 } }`, nil, "headers content should be key/value strings"},
                // Positive tests
                {"listen port only", `websocket { listen: 1234 }`, func(wo *WebsocketOpts) error {
                        if wo.Port != 1234 {
@@ -1636,7 +1638,8 @@ func TestWSParseOptions(t *testing.T) {
                                }
                        }
                        `, func(wo *WebsocketOpts) error {
-                               if wo.Headers != "Strict-Transport-Security: max-age=15768000 ; includeSubDomains\r\nX-Frame-Options: DENY\r\n" {
+                               if !strings.Contains(wo.Headers, "Strict-Transport-Security: max-age=15768000 ; includeSubDomains\r\n") &&
+                                       !strings.Contains(wo.Headers, "X-Frame-Options: DENY\r\n") {
                                        return fmt.Errorf("Invalid headers: %v", wo.Headers)
                                }
                                return nil
```